server/config.go

```diff
@@ -112,7 +114,7 @@ func (c *Configuration) AddDomain(domain string) (err error) {
 }
 
 // NewConfig will make a new server configuration
-func NewConfig(serviceProvider PaymailServiceProvider, opts ...ConfigOps) (*Configuration, error) {
+func NewConfig(serviceProvider *PaymailServiceLocator, opts ...ConfigOps) (*Configuration, error) {
```

Comment:
Maybe it would be beneficial to note here (in a func comment) that appropriate services should be registered BEFORE calling it. 

Especially because `GetPikeService` will throw panic if PIKE service has not been registered.